### ISSUE TITLE
Ensure no duplicate players

### ISSUE THREAD

**Issue by matthewkirby** [matthewkirby/RSLLeaderboard#10]
It is possible to have duplicate players due to typos while submitting rated async information or for a new racetime.gg patreon supporter. Enable quick checks to look for this. Due to the low number of unique users in this project, this is probably doable by just checking racetime.gg name without the discriminator.

**Comment by fenhl** [comment]
Each racetime.gg user has a unique ID. This ID is static even if the user changes their display name or becomes a Patreon supporter. So you should be able to keep track of users via that ID.

**Comment by matthewkirby** [comment]
Yeah this is what I use to track players. The problems arise with the rated asyncs. Currently, we just ask for a link to your racetime profile since the urls end in the user id. This work unless, say, someone somehow omits the last character from the url 🙃 (that happened last week) or patreon supporters who's url becomes their chosen name and then I get an entry for their user id (from racetime races) and their racetime name.

The long-term plan is to have the entire rated async process through RSLBot. That would have you register your racetime profile with the bot and then it always knows who you are (and solve the first case, which is much more common than someone suddenly becoming a patreon supporter). I expect there is also potentially a way to get the userid from a racetime api endpoint but I just havent looked into it yet.

**Comment by matthewkirby** [comment]
Looking at the api, there is an endpoint to search for users. But that doesn't exactly solve the problem either since profile url doesn't correspond to chosen name if you get an idiot like me: https://racetime.gg/user/rando-god

**Comment by fenhl** [comment]
You can append `/data` to the profile URL (e.g. <https://racetime.gg/user/rando-god/data>) to get the user data including the ID.

**Comment by matthewkirby** [comment]
Oh! This is perfect! I think I tried this but forgot the /user/ part haha.
Thanks a ton!

**Comment by fenhl** [comment]
> The long-term plan is to have the entire rated async process through RSLBot. That would have you register your racetime profile with the bot and then it always knows who you are (and solve the first case, which is much more common than someone suddenly becoming a patreon supporter).

Would it make sense to use Mido's House for this? I already have a mapping between racetime.gg and Discord accounts, and could set up an API that the leaderboard can use to look them up. This would also verify that people are specifying their own racetime.gg account and not someone else's, since I'm using OAuth.

**Comment by matthewkirby** [comment]
Hmm yeah, that might be good. This is still somewhat long term. My first goal is to replicate the rsl-leaderboard.web.app site with all of its features along with some super basic RSLBot integration to, say, manually update the leaderboard or tell it to run the script to process the rated async data. 

In this scenario, you would just have an api that I could send the discord account info and it would send back the racetime userid?

**Comment by fenhl** [comment]
Exactly. Of course it would only work for people who have signed in with both on MH. So if that's not the case, RSLBot would have to generate an error reply telling the user how to do that.
